graylog2-server/src/test/java/org/graylog2/telemetry/rest/TelemetryServiceTest.java

```diff
@@ -0,0 +1,166 @@
+import org.graylog2.system.traffic.TrafficCounterService;
+import org.graylog2.telemetry.db.DBTelemetryUserSettingsService;
+import org.graylog2.telemetry.db.TelemetryUserSettingsDto;
+import org.graylog2.telemetry.enterprise.TelemetryEnterpriseDataProvider;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.graylog2.system.traffic.TrafficCounterService.TrafficHistogram;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class TelemetryServiceTest {
+
+    public static final TrafficHistogram TRAFFIC_HISTOGRAM = TrafficHistogram.create(
+            DateTime.now(DateTimeZone.UTC), DateTime.now(DateTimeZone.UTC), Map.of(), Map.of(), Map.of());
+    private static final String CURRENT_USER = "current_user";
+    private static final String USER_TELEMETRY_SETTINGS = "user_telemetry_settings";
+    private static final String CLUSTER = "cluster";
+    private static final String LICENSE = "license";
+    private static final String PLUGIN = "plugin";
+    private static final String SEARCH_CLUSTER = "search_cluster";
+    @Mock
+    TrafficCounterService trafficCounterService;
+    @Mock
+    ClusterConfigService clusterConfigService;
+    @Mock
+    TelemetryEnterpriseDataProvider enterpriseDataProvider;
+    @Mock
+    UserService userService;
+    @Mock
+    Set<PluginMetaData> pluginMetaDataSet;
+    @Mock
+    ClusterAdapter elasticClusterAdapter;
+    @Mock
+    SearchVersion elasticsearchVersion;
+    @Mock
+    ObjectMapper objectMapper;
+    @Mock
+    DBTelemetryUserSettingsService dbTelemetryUserSettingsService;
+    @Mock
+    EventBus eventBus;
+    @Mock
+    User user;
+
+    @Test
+    void test_telemetry_is_disabled_globally() {
+        TelemetryService telemetryService = createTelemetryService(false);
+        mockUserTelemetryEnabled(true);
+
+        Map<String, Object> response = telemetryService.getTelemetryResponse(user, Map.of());
+
+        assertThat(response).containsOnlyKeys(USER_TELEMETRY_SETTINGS);
+    }
+
+    @Test
+    void test_telemetry_is_disabled_for_user() {
+        TelemetryService telemetryService = createTelemetryService(true);
+        mockUserTelemetryEnabled(false);
+
+        Map<String, Object> response = telemetryService.getTelemetryResponse(user, Map.of());
+
+        assertThat(response).containsOnlyKeys(USER_TELEMETRY_SETTINGS);
+    }
+
+    @Test
+    void test_no_telemetry_user_settings() {
+        TelemetryService telemetryService = createTelemetryService(true);
```

Comment:
Why do telemetry data have to be collected when the permission has not been asked?

Reply:
We do not check permission here, as a user can only check his own telemetry settings.